yarn-project/pxe/src/kernel_prover/kernel_prover.ts

```diff
@@ -158,22 +125,33 @@ export class KernelProver {
         pushTestData('private-kernel-inputs-inner', proofInput);
         output = await this.proofCreator.createProofInner(proofInput);
       }
-      (await this.getNewNotes(currentExecution)).forEach(n => {
-        newNotes[n.commitment.toString()] = n;
-      });
       firstIteration = false;
       previousVerificationKey = privateCallData.vk;
     }
 
     const previousVkMembershipWitness = await this.oracle.getVkMembershipWitness(previousVerificationKey);
-    const previousKernelData = new PrivateKernelInnerData(
+    const previousKernelData = new PrivateKernelData(
       output.publicInputs,
       output.proof,
       previousVerificationKey,
       Number(previousVkMembershipWitness.leafIndex),
       assertLength<Fr, typeof VK_TREE_HEIGHT>(previousVkMembershipWitness.siblingPath, VK_TREE_HEIGHT),
     );
 
+    const readNoteHashHints = this.hintsBuilder.getNoteHashReadRequestHints(
```

Comment:
Moved these from after ordering to before ordering, to make it clear that they are operated on un-sorted data.